sdk/keyvault/Azure.Security.KeyVault.Keys/api/Azure.Security.KeyVault.Keys.netstandard2.0.cs

```diff
@@ -333,11 +359,21 @@ public partial class DecryptResult
         public static bool operator !=(Azure.Security.KeyVault.Keys.Cryptography.EncryptionAlgorithm left, Azure.Security.KeyVault.Keys.Cryptography.EncryptionAlgorithm right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class EncryptOptions
+    {
+        public EncryptOptions(byte[] plaintext, byte[] iv = null) { }
```

Comment:
Since `iv` is optional should we make the property settable vs. setting via the ctor?

Reply:
/cc @KrzysztofCwalina because this is one I debated in my head a bit...

My thinking was that, while it's optional, it's only optional in some cases so putting it in the constructor helps hoist it to visibility. That said, I could do the same as I did for `DecryptOptions`, which was have overloads where - not only do I get a chance to document it*, but also can make sure that *that* ctor requires it (i.e. non-null).

\* That reminds me, I need to push an update with better docs to say which ctors to use with which algorithms. I'll wait for any of your additional feedback before I push, though.

Reply:
I was just about to change this and remembered why (given the current XML doc comment): yes, it can be null even when needed; however, in that case we will generate it. Making it an optional param gives me opportunity for a comment like so:
```c#
/// <param name="iv">
/// The initialization vector for encryption. If null, one will be generated for symmetric algorithms like an
/// nonce and returned in <see cref="EncryptResult.Iv"/>.
/// </param>
```
Yes, it could be done with two ctors but, the fact it's still optional in the other case would seem to obviate the need to have the overload. The overload that takes it could require it, but then people might think they are responsible for generating the 96-bit nonce for AES-GCM.

Reply:
I'm assuming we generate in the client and not in the options class. If that's the case it seems we could just put this doc comment onto the property rather the the ctor param no?

Reply:
But it can also be passed in, and should when not used as an nonce.

That said, @KrzysztofCwalina agreed offline to factory methods like so, that will make these cases more clear:
```c#
EncryptResult Encrypt(EncryptOptions.A128CbcOptions(byte[] plaintext, byte[] iv), CancellationToken);
```

Reply:
I like this approach. I'll approve this PR if we need to merge as is for a preview, but I think the approach you have with factory methods above is the best.